### ISSUE TITLE
Tooltip pattern should allow for aria-labelledby

### ISSUE THREAD

**Issue by carmacleod** [w3c/aria-practices#1034]
Opening a separate issue to address @Yaffle's https://github.com/w3c/aria-practices/issues/128#issuecomment-352376427 about icon toolbar buttons having tooltips that provide their label.

The current [Tooltip pattern](https://w3c.github.io/aria-practices/#tooltip) does not allow for this. It says:
> The element that triggers the tooltip references the tooltip element with aria-describedby.

Need to add some words to the effect that the author must choose between referencing the tooltip element with `aria-labelledby` or with `aria-describedby`. Something like, "If it doesn't have a name, then the tooltip element should contain a concise name string, and should be referenced with `aria-labelledby`. It it already has a name and the tooltip is providing a longer description, then use `aria-describedby`."

**Comment by scottaohara** [comment]
@carmacleod per my comment from the closed tooltip thread https://github.com/w3c/aria/issues/987#issuecomment-497992780, and the further discussion in aria 979, is this issue still being considered?

I saw @ZoeBijl's tooltip apg example on twitter today which also used a `tooltip` as the name for a button, which reminded me that there were issues about this and i recalled trying to make sure we _didn't_ promote something like that with apg.

Have thoughts on that changed?   Or has the covid years just clouded my memory into assuming that this topic was put to rest?

**Comment by ZoeBijl** [comment]
> which reminded me that there were issues about this
Do you recall what those issues were? My (limited) testing didn’t reveal any major issues.

**Comment by scottaohara** [comment]
by "issues" I meant the 2 GitHub issues filed against ARIA concerning `tooltip` and the discussions about _not_ using `role=tooltip` when providing a name for a control.  

That's not how the role is spec'd right now.  That doesn't mean it couldn't change... I would love for the role to change (or be killed off since it's largely useless) ...  but it'd be great to resolve that before providing a pattern with APG that doesn't adhere to the ARIA definition.  

Per the test case you have, the only "issue" - in this case bug/quirk - I came across was that I can access the "tooltip" separate from the component.  Which, is not really a big deal imo with the descriptive tooltip... but is odd to navigate to a button, hear its accessible name, and then navigate again and hear the name again as static text.  I provided a screenshot/example about this in [the linked ARIA issue](https://github.com/w3c/aria/issues/979#issuecomment-497994552).

**Comment by ZoeBijl** [comment]
Ah yes. I agree that `role=tooltip` doesn’t make much sense in the [“Tooltip as main label” example](https://zoebijl.github.io/apg-tooltip/#tooltip-main-label). Perhaps the example could be updated to say “Don’t add `role=tooltip` in these cases:”?
